app/src/main/java/com/tehmou/rxbookapp/viewmodels/RepositoriesViewModel.java

```diff
@@ -0,0 +1,96 @@
+package com.tehmou.rxbookapp.viewmodels;
+
+import com.tehmou.rxbookapp.RxBookApp;
+import android.util.Log;
+
+import com.tehmou.rxbookapp.data.DataLayer;
+import com.tehmou.rxbookapp.pojo.GitHubRepository;
+import com.tehmou.rxbookapp.pojo.GitHubRepositorySearch;
+import com.tehmou.rxbookapp.view.RepositoriesView;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Objects;
+import java.util.concurrent.TimeUnit;
+
+import javax.inject.Inject;
+
+import rx.Observable;
+import rx.functions.Func1;
+import rx.functions.FuncN;
+import rx.subjects.BehaviorSubject;
+import rx.subjects.PublishSubject;
+import rx.subjects.Subject;
+import rx.subscriptions.CompositeSubscription;
+
+
+/**
+ * Created by ttuo on 19/03/14.
+ */
+public class RepositoriesViewModel {
+    private static final String TAG = RepositoriesViewModel.class.getSimpleName();
+    final private CompositeSubscription compositeSubscription = new CompositeSubscription();
+
+    @Inject
+    DataLayer dataLayer;
+
+    private PublishSubject<Observable<String>> searchString = PublishSubject.create();
+
+    final private Subject<List<GitHubRepository>, List<GitHubRepository>> repositories
+            = BehaviorSubject.create();
+
+    public Observable<List<GitHubRepository>> getRepositories() {
+        return repositories;
+    }
+
+    public RepositoriesViewModel() {
+        RxBookApp.getInstance().getGraph().inject(this);
+        Log.v(TAG, "RepositoriesViewModel");
+    }
+
+    public void subscribeToDataStore() {
+        Log.v(TAG, "subscribeToDataStore");
+
+        compositeSubscription.add(
```

Comment:
You are subscribing and unsubscribing several times from it. And CompositeSubscription has to be recreated then. 

http://reactivex.io/RxJava/javadoc/rx/subscriptions/CompositeSubscription.html#clear()